### ISSUE TITLE
copy size-checks of pull-requests from pyqtgraph

### ISSUE THREAD

**Issue by ericdill** [scikit-beam/scikit-beam#420]
It is possible to accidentally commit large files.  If you commit a large file by accident and then delete the large file and make another commit, *that large file is still part of the repo!*  We do not check for this currently.  We should check for this.  @campagnola has some nice size checks on travis in the pyqtgraph project that we should steal.

**Comment by souravsingh** [comment]
Are we referring to [this test](https://github.com/pyqtgraph/pyqtgraph/blob/develop/.travis.yml#L153) for size check?

**Comment by ericdill** [comment]
Yup

**Comment by ericdill** [comment]
Thanks for pushing this along @souravsingh .  I made some changes and tested on travis-ci for the ericdill/scikit-beam fork.  This PR got closed automatically, as I merged locally and pushed to master (normally don't do this, but I rebased a few times to get rid of the extra merge commits in this PR).  Thanks!  (note that your commits other than the merge commits are in master now: https://github.com/scikit-beam/scikit-beam/commits/master)  :smile:
